code/modules/mafia/controller.dm

```diff
@@ -667,31 +697,81 @@
 		. += L[key]
 
 /**
-  * Returns all setups that the amount of players signed up could support (so fill each role)
-  * Arguments:
-  * * ready_count: the amount of players signed up (not sane, so some players may have disconnected or rejoined ss13).
+  * Returns a semirandom setup, with...
+  * Town, Two invest roles, one protect role, sometimes a misc role, and the rest assistants for town.
+  * Mafia, 2 normal mafia and one special.
+  * Neutral, two disruption roles, sometimes one is a killing.
+  *
+  * Assistants must be filled in their own list before merging the two because of how byond dicts work
+  * See _defines.dm in the mafia folder for a rundown on what these groups of roles include.
+  */
+/datum/mafia_controller/proc/generate_random_setup()
+	var/invests_left = 2
+	var/protects_left = 1
+	var/miscs_left = prob(35) ? 1 : 0
+	var/mafiareg_left = 2
+	var/mafiaspe_left = 1
+	var/killing_role = prob(50) ? 1 : 0
+	var/disruptors = killing_role ? 1 : 2 //still required to calculate overflow
```

Comment:
I hate to say it but if you're gonna use them as booleans why not just do var/1or0 = prob(number)

Reply:
it may be a bit better for reading comprehension but actually i agree doing just prob for these is much better